### PR TITLE
fix: 🐛 Fix voting problems

### LINKED ISSUE #117: Fix Voting Bugs
Fix voting bugs because catniss wants a smooth and unbroken gameplay experience.

Tasks:

- [x] Display voting as a heading
- [x] Show prompt answers during voting 
- [x] Dead players should be greyed out during the voting phase
- [x] Dead players shouldn't be able to vote
- [x] Players should be able to vote for themselves
- [x] Highlight who you voted for during the voting phase
- [x] Mobile responsive

Acceptance Criteria:
- Display voting as a heading
- Show prompt answers during voting 
- Dead players should be greyed out during the voting phase
- Dead players shouldn't be able to vote
- Players should be able to vote for themselves
- Highlight who you voted for during the voting phase
- Mobile responsive

### PR DESCRIPTION
**Pull request by gchamb**
#117

### CODE REVIEW

File: client/src/components/Vote.svelte [hunk 1]
```diff
@@ -36,36 +37,46 @@
...
+            await addVote(lobbyCode, $user?.uid ?? "", lobby.uids[i]);
+            selected = i;
+          }}
         >
-          <img src="/avatars/{avatar}.webp" alt={avatarAltText[avatar]} />
+          <img class={!alive ? "dead" : ""} src="/avatars/{avatar}.webp" alt={avatarAltText[avatar]} />
           <span class="mdc-typography--subtitle1">{displayName ?? ""}</span>
+          <div class="mdc-typography--caption">
+            {#if alive}
+              {promptAnswer ?? "no answer"}
```
Comment (mrjoshua520) [inline client/src/components/Vote.svelte]: For living players prompt answer here, maybe we want to indicate it somehow in addition to the smaller text. I'm thinking putting something like "Answer: their_answer" or maybe changing the color of the text to something other than the default

**Review by mrjoshua520 — CHANGES_REQUESTED** [review]

**Review by mrjoshua520 — DISMISSED** [review]

File: firestore.rules [hunk 2]
```diff
@@ -63,15 +66,14 @@ service cloud.firestore {
         function checkTarget(){
           return request.resource.data.keys().hasOnly(['target'])
           && request.resource.data.target is string
-          && request.resource.data.target in lobbyData().uids
-          && request.resource.data.target != request.auth.uid;
+          && request.resource.data.target in lobbyData().alivePlayers
```
Comment (nstringham) [inline firestore.rules:69]: ;

File: functions/src/lobby.ts [hunk 3]
```diff
@@ -195,6 +196,18 @@ export const leaveLobby = functions.https.onCall((data: unknown, context): Promi
...
+  const lobbyDocRef = change.after.ref as firestore.DocumentReference<Lobby>;
+  const lobby = change.after.data() as Lobby;
+  const alivePlayers = [];
+  for (let i = 0; i < lobby.uids.length; i++) {
+    if (lobby.players[i].alive) {
+      alivePlayers.push(lobby.uids[i]);
+    }
+  }
+  lobbyDocRef.update({ alivePlayers });
+});
```
Comment (nstringham) [inline functions/src/lobby.ts]: infinite loop stuff

File: functions/src/lobby.ts [hunk 4]
```diff
@@ -51,6 +51,7 @@ export const createLobby = functions.https.onCall(async (data: unknown, context)
       },
     ],
     state: "WAIT",
+    alivePlayers: [],
```
Comment (nstringham) [inline functions/src/lobby.ts]: should include context.auth.uid

File: firestore.rules (hunk 2 above)
Comment (nstringham) [inline firestore.rules]: ;

File: client/src/components/Vote.svelte [hunk 5]
```diff
@@ -36,36 +37,46 @@
...
-          on:click={() => addVote(lobbyCode, $user?.uid ?? "", lobby.uids[i])}
+          class="avatar {!alive ? 'dead' : ''} {selected == i ? 'selected' : ''}"
+          disabled={!alive}
+          on:click={async () => {
+            await addVote(lobbyCode, $user?.uid ?? "", lobby.uids[i]);
+            selected = i;
+          }}
         >
-          <img src="/avatars/{avatar}.webp" alt={avatarAltText[avatar]} />
+          <img class={!alive ? "dead" : ""} src="/avatars/{avatar}.webp" alt={avatarAltText[avatar]} />
```
Comment (nstringham) [inline client/src/components/Vote.svelte]: why not just use the parrent?

File: client/src/components/Vote.svelte [hunk 6]
```diff
@@ -36,36 +37,46 @@
...
+          <div class="mdc-typography--caption">
+            {#if alive}
+              Answer: {promptAnswer ?? "no answer"}
+            {:else}
+              Im dead
+            {/if}
+          </div>
         </button>
-        <span class="mdc-typography--heading6">{votes ?? 0}</span>
+        <span class="mdc-typography--heading6 {!alive ? 'dead' : ''}">{votes ?? 0}</span>
```
Comment (nstringham) [inline client/src/components/Vote.svelte]: why not just use the parrent?

File: client/src/components/Vote.svelte [hunk 7]
```diff
@@ -93,4 +104,67 @@
...
+      background-position: center;
+      background-repeat: no-repeat;
+    }
+  }
+
+  .dead {
+    opacity: 0.5;
+  }
+  .selected {
+    border-color: green;
```
Comment (nstringham) [inline client/src/components/Vote.svelte]: use the theme color

**Review by nstringham — DISMISSED** [review]

File: client/src/components/Vote.svelte [hunk 8]
```diff
@@ -36,36 +54,43 @@
...
         </button>
         <span class="mdc-typography--heading6">{votes ?? 0}</span>
       </div>
     {/each}
   </div>
 </div>
 
-<style>
+<style lang="scss">
   .voting {
```
Comment (nstringham) [inline client/src/components/Vote.svelte:83]: Why did you make it scss?

File: functions/src/lobby.ts (hunk 3 above)
Comment (nstringham) [reply functions/src/lobby.ts]: good enough for now

**Review by nstringham — APPROVED** [review]

**Pull request merged**